### ISSUE TITLE
Modify Inputs and Outputs Independently Using Optics

### ISSUE THREAD

**Issue by solomon-b** [cofree-coffee/cofree-bot#21]

**Comment by solomon-b** [comment]
I can't remember the context for this. Is it something other then `dimap`? @masaeedu

**Comment by masaeedu** [comment]
I don't remember what this specific issue is about either, but assuming we have various subclasses of `Profunctor` implemented, and we do a little busywork to deal with the annoying historical presentational warts of profunctor optics in Haskell, we should be able to "focus" existing bots and servers by simply applying profunctor optics to them.

**Comment by solomon-b** [comment]
I agree that this is at its core a solved issue by our basic design. I'm going to close the issue. If anything more specific that we cannot do reveals itself we can make a fresh issue.
